ui/app/src/services/configuration.ts

```diff
@@ -258,67 +197,77 @@ function deserializeActiveTargetingModelData<T extends Object>(data: T, contentT
-            `Filtered out config item with blank/null width/height values. ` +
-            `Both values in blank is equivalent to the tool's default preset.`
-          );
-          return false;
-        } else {
-          return true;
-        }
-      })
-    };
-  }
-}
-
 // endregion
 
 // region SidebarConfig
 
-export interface SidebarConfigItem {
-  name?: string;
-  params?: object;
-  render?: string;
-  props?: object;
-}
+export function getSiteUiConfig(site: string): Observable<any> {
+  const widgetParser = (items): SidebarPanelConfigEntry[] => {
+    let array = asArray(items.widget);
+    return array.map((item) => ({
+      id: item.id,
+      ...(item.roles?.role && { roles: asArray(item.roles.role) }),
+      ...(item.parameters && {
+        parameters: {
+          ...item.parameters,
+          ...(item.parameters.excludes && { excludes: asArray(item.parameters.excludes.exclude) })
+        }
+      })
+    }));
+  };
 
-export function getSidebarItems(site: string): Observable<SidebarConfigItem[]> {
-  return getRawConfiguration(site, '/context-nav/sidebar.xml', 'studio').pipe(
-    map((rawXML) => {
-      // The XML has a structure like:
-      // {root}.contextNav.contexts.context.groups.group.menuItems.menuItem.modulehooks.modulehook;
-      // To avoid excessive traversal, create a transition XML document with just the items.
-      if (rawXML) {
-        const items = Array.from(fromString(rawXML).querySelectorAll('modulehook'));
-        const cleanDoc = fromString(`<?xml version="1.0" encoding="UTF-8"?><root></root>`);
-        cleanDoc.documentElement.append(...items);
-        return asArray<SidebarConfigItem>(toJS(cleanDoc).root.modulehook).filter(Boolean);
+  const panelsParser = (items): SidebarPanelConfigEntry[] => {
+    let array = asArray(items.panel);
+    return array.map((item) => ({
+      id: item.id,
+      ...(item.roles?.role && { roles: asArray(item.roles.role) }),
+      ...(item.parameters && {
+        parameters: {
+          ...item.parameters,
+          ...(item.parameters.widgets && { widgets: widgetParser(item.parameters.widgets) }),
+          ...(item.parameters.devices && { devices: asArray(item.parameters.devices.device) })
+        }
+      })
+    }));
+  };
+
+  const linksParser = (items): SiteNavConfigEntry[] => {
+    let array = asArray(items.link);
+    return array.map((item) => ({
+      ...item,
+      ...(item.roles?.role && { roles: asArray(item.roles.role) })
+    }));
+  };
+
+  return getConfigurationDOM(site, '/ui.xml', 'studio').pipe(
+    map((xml) => {
+      if (xml) {
+        const parsed = deserialize(xml).ui;
+        return {
+          preview: {
+            sidebar: {
+              panels: panelsParser(parsed.preview.sidebar.panels)
+            },
+            siteNav: {
+              links: linksParser(parsed.preview.siteNav.links)
+            }
+          }
+        };
       } else {
         return [];
```

Comment:
Incorrect.

In fact, for now, if the file is missing, while we have a UM, return a mockup with the typical values.